pkg/utils/utils.go

```diff
@@ -255,3 +256,45 @@ func StringMapKeys(m interface{}) []string {
 	sort.Strings(keys)
 	return keys
 }
+
+// Optional returns the first optional non-zero element given as variadic argument,
+// if given, or the zero element as default.
+func Optional[T any](list ...T) T {
+	var zero T
+	for _, e := range list {
+		if !reflect.DeepEqual(e, zero) {
+			return e
+		}
+	}
```

Comment:
That changes the original logic, is it intentional?

* Original logic: Return the first entry if we have at least one
* New logic: Return with the first non-zero entry

We get different result from:
* `[nil, nil, nil, *obj]`
  * old return value: `nil`
  * new return value: `*obj`
* `[Data{}, Data{Name: "name"}, Data{Name: "name2"}]`
  * old return value: `Data{}`
  * new return value: `Data{Name: "name"}`
* `[false, true, true, false]`
  * old return value: `false`
  * new return value: `true`

Reply:
yes, that was a problem with these various implementations, every one worked slightly different. Now I've aligned it, and it works the same way everywhere.

Reply:
The only real difference is how booleans are handled. Previously if the first entry in a list of booleans was a false, the return value was false.